src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -249,18 +255,31 @@ private boolean areTotalsEqual(InvoiceLine invoiceLine1, InvoiceLine invoiceLine
    * 4. Update corresponding Invoice
    * @param id invoiceLine id to be deleted
    */
-  public CompletableFuture<Void> deleteInvoiceLine(String id) {
-    String query = QUERY_PARAM_START_WITH + id;
-
-    return getInvoices(query, httpClient, ctx, okapiHeaders, logger, lang)
-      .thenApply(invoiceCollection -> invoiceCollection.getInvoices()
-        .get(0))
+  public CompletableFuture<Void> deleteInvoiceLine(String lineId) {
+    return getInvoicesIfExists(lineId).thenApply(invoiceCollection -> invoiceCollection.getInvoices()
+      .get(0))
       .thenCompose(invoice -> protectionHelper.isOperationRestricted(invoice.getAcqUnitIds(), DELETE)
         .thenApply(vvoid -> invoice))
-      .thenCompose(invoice -> handleDeleteRequest(resourceByIdPath(INVOICE_LINES, id, lang), httpClient, ctx, okapiHeaders, logger)
+      .thenCompose(invoice -> handleDeleteRequest(resourceByIdPath(INVOICE_LINES, lineId, lang), httpClient, ctx, okapiHeaders, logger)
         .thenRun(() -> updateInvoiceAndLinesAsync(invoice)));
   }
 
+  private CompletableFuture<InvoiceCollection> getInvoicesIfExists(String lineId) {
+    String query = QUERY_PARAM_START_WITH + lineId;
+    return getInvoices(query, httpClient, ctx, okapiHeaders, logger, lang).exceptionally(t -> {
+      Throwable cause = t.getCause();
+      // When specified invoice line does not exist
+      if (cause instanceof HttpException && ((HttpException) cause).getCode() == Response.Status.NOT_FOUND.getStatusCode()) {
+        List<Parameter> parameters = Collections.singletonList(new Parameter().withKey("invoiceLineId")
+          .withValue(lineId));
+        Error error = INVOICE_NOT_FOUND.toError()
```

Comment:
I think it may be confusing for the user to get an `invoiceNotFound` error message when trying to delete the invoice line.